tests/bitswap.rs

```diff
@@ -0,0 +1,180 @@
+use blockstore::{Blockstore, InMemoryBlockstore};
+use futures::{future::FutureExt, poll};
+use tokio::time::{sleep, Duration};
+
+use crate::utils::{cid, spawn_node};
+
+mod utils;
+
+#[tokio::test]
+async fn test_client_request() {
+    let data = "foo";
+    let cid = cid(data.as_bytes());
+    let store = InMemoryBlockstore::new();
+    store.put_keyed(&cid, data.as_ref()).await.unwrap();
+
+    let server = spawn_node(Some(store)).await;
+    let mut client = spawn_node(None).await;
+
+    drop(client.connect(&server));
+    let received = client.request_cid(cid).await.expect("could not get CID");
+
+    assert_eq!(&received[..], data.as_bytes());
+}
+
+#[tokio::test]
+async fn test_server_request() {
```

Comment:
this looks like the exact same test as above :smiley:

Reply:
almost - in both cases single "client" connects to a single "server", but then either server or client requestes data over bitswap. This should go through virtually the same code, just wanted to do a basic check that there's no weird discrepancy in behaviour, depending on who's initiating the connection.

Reply:
I'm not sure that test is useful but leaving that up to you